### DISCUSSION TITLE
[Bug]: frequent write access to sd card (every second)

### DISCUSSION THREAD

**Discussion by Joerg76** [zwave-js/zwave-js-ui#2288] [Q&A]
### Checklist

- [X]  I am not using Home Assistant. Or: a developer has told me to come here.
- [X] I have checked the troubleshooting section and my problem is not described there.
- [X] I have read the changelog and my problem is not mentioned there.

### Deploy method

Docker

### Zwavejs2Mqtt version

6.5.2

### ZwaveJS version

8.11.6

### Describe the bug

My zwave network consists of only 3 sensors and 3 relays, but data is still written to the SD card every second. The modification date of the directories ....metadata.jsonl.lock and ....values.jsonl.lock changes every second. This behavior should be changed for the durability of the SD card. 

### To Reproduce

docker with -v /yourpath:/usr/src/app/store and monitor the modification date 

### Expected behavior

The lock status should be saved in main memory and only saved to disk when the data changes.

**Comment by kpine** [comment]
FYI, the lockfile is managed by the driver, not zwavejs2mqtt, and you can [set](https://zwave-js.github.io/node-zwave-js/#/api/driver?id=zwaveoptions) the `ZWAVEJS_LOCK_DIRECTORY` environment variable to point to another directory that isn't on your SD card, if this worries you.

It's not necessarily the case that the lockfile updates are translating to SD card writes. I believe the updates to the lockfiles are just changing the file mtime. Sometimes this only involves a memory operation. For example, the root filesystem for my RPi is configured with ext4 mount option `lazytime`, which means mtime updates are only in memory for the most part. You might take a look at `lazytime` if you're not using it and have the option to.

Still, I opted to set the lock directory for my installation to a directory that is on tmpfs. In my docker compose file I have:

```yaml
    environment:
      - "ZWAVEJS_LOCK_DIRECTORY=/run/lock/zwave-js"
    ...
    volumes:
      - /run/lock/zwave-js:/run/lock/zwave-js
    ...
```
`/run/lock` on the host is a tmpfs.
```
$ mount | grep /run/lock
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
```

You would still want the lock directory to be stored on the host somewhere to prevent multiple docker instances from accessing the cache files at the same time.

**Comment by Joerg76** [comment]
Thank you for the informations.
